### PR TITLE
Distribution with parameters. Editing with user controls

### PR DESCRIPTION
**Pull request by Lunalo**

### CODE REVIEW

**Comment by dannyparsons** [comment]
@lilyclements could you have a test of this?
I think we can fix the conflict.

**Comment by lilyclements** [comment]
I think that we also will have a conflict later with another two files - One Var Fit Model and Regression Simple where I have changed some dialogs.
I've tested it and it crashes whenever I try to change the distribution
It takes you to line 554 of ucrDistributions.
            clsCurrDistribution = lstCurrentDistributions(ucrInputDistributions.cboInput.SelectedIndex)

**Comment by lilyclements** [comment]
An additional point - since I know how One Var Fit Model and Regression Simple work fairly well, and I know when they work etc, I made an arbitrary branch off of the OneVarFitModel branch and added the above changes to it. I then looked at how OneVarFitModel changed

If I now change from general to exact (i.e., from displaying any distribution to restricting to four - Normal, Poisson, Binomial and None) the crash takes me to ucrDistributions, line 177
                ucrInputDistributions.cboInput.Items.Add(translate(Dist.strNameTag))
Upon pressing continue, it takes me to the main form where we have .ShowDialog() line for this dialog
Upon pressing continue again, I am able to now look at the page with 'Exact' selected. We should be restricted to the four aforementioned distributions, however, this is not the case and instead we are given all distributions
If I try to select a distribution, it crashes and takes me to line 554 - mentioned in my previous comment

Finally, a very small thing but the ucrdistributions size has now changed, as a result, we need to increase the size of it on any dialogs involving this ucr.

Just a few things to note anyway!

**Comment by Lunalo** [comment]
Hi Lily,

I have looked at the errors and I think they are majorly caused by the those two dialogs other than the conrol itself. 

I did the first and second selection of the distribution which was fine. When I did the third selection, It then breaks  to line 554 in the ucrDistribution.
When I add watch then the selected index is different from the list index . I think you can help solve this since you were actively involved in the adding of these distributions. 

Thanks.

**Comment by lilyclements** [comment]
I'll have a look into it. However, it did crash on other dialogs - for example, on dlgRandomSample, if I select 'Students T' it crashed ('Poisson' didn't crash, I didn't test any others but I can)

**Comment by lilyclements** [comment]
I think to fix the issue I mentioned above in to in Line 139 put
        ucrInputDistributions.cboInput.Items.Clear()
instead of .Reset()

File: instat/ucrDistributions.vb [hunk 1]
```diff
@@ -136,7 +136,8 @@ Public Class ucrDistributions
     Public Sub SetDistributions()
         Dim bUse As Boolean
         lstCurrentDistributions.Clear()
-        ucrInputDistributions.Reset()
+        ucrInputDistributions.cboInput.Items.Clear()
```
Comment (dannyparsons) [inline instat/ucrDistributions.vb]: This is right, but there should be a Reset method which does this line. So that here we can call Reset and it works.

**Comment by Lunalo** [comment]
@dannyparsons 

The current reset method in the cboinput is doing something different from clearing the items. We understand both ucrinputcombo and ucrinputtextbox are inheriting from it. I have added a reset method into the combo for clearing.

**Comment by dannyparsons** [comment]
You don't need to edit the combo box. All you need to do is clear the items in the reset method of the distributions control.

**Comment by Lunalo** [comment]
Then i can do something like this in the ucrCombo?

    Public Overrides Sub Reset()
        cboInput.Items.Clear()
    End Sub

File: instat/ucrInputComboBox.vb [hunk 2]
```diff
@@ -146,7 +146,7 @@ Public Class ucrInputComboBox
         bUserTyped = True
     End Sub
 
-    Public Sub ResetCombo()
+    Public Overrides Sub Reset()
         cboInput.Items.Clear()
     End Sub
```
Comment (dannyparsons) [inline instat/ucrInputComboBox.vb]: It's not clear we want this for all combo boxes. For most of them we don't want to clear the items. Instead just add that line in the reset of the distributions control.

Comment (Lunalo) [inline instat/ucrInputComboBox.vb]: There is no reset  method for ucrDistributions so  i have added it

**Comment by Lunalo** [comment]
This is also ready

**Comment by dannyparsons** [comment]
@lilyclements are you able to test now?

**Comment by lilyclements** [comment]
Not yet! The new laptop should be coming at some point today (however I have to leave at 330/4), so if I get the laptop by 330/4 then I can download everything later. Otherwise, I'll pop into the office after the kids (around 10pm) to collect the laptop and download everything overnight.

**Comment by lilyclements** [comment]
Was the plan to accept dlgRegressionHyp and dlgOneVarFitModel before merging this, however, so all conflicts are limited to one branch?

**Comment by dannyparsons** [comment]
Yes I can accept those ones first.

**Comment by lilyclements** [comment]
This is okay. It currently disables the receiver before resetting on ones like dlgThreeVariableModelling, so then it is disabled on the chosen distribution rather than the default.

Also, some do not reset at all - such as Random Sample.

Other than that it works well!

**Comment by maxwellfundi** [comment]
@lilyclements @Lunalo  what the state of this? Can @dannyparsons look at yet?

**Comment by lilyclements** [comment]
Since my last review, there doesn't seem to be any commits so I am waiting for the bugs to be sorted before Danny can review it

**Comment by Lunalo** [comment]
@lilyclements 

I think we need to have a discussion on this

**Comment by lilyclements** [comment]
How is this going now?

**Comment by stevekogo** [comment]
@Lunalo It seems a lot of work was done here. Comment on its state.

**Comment by Lunalo** [comment]
@stevekogo 

There is still a lot to be done. There was still a problem with one event .  So, I am still working on this

**Comment by dannyparsons** [comment]
I assume this can be closed?

**Pull request closed without merging**